frontend/__tests__/plans-edit.test.tsx

```diff
@@ -0,0 +1,166 @@
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { Toaster } from "sonner";
+import { afterEach, beforeEach, describe, expect, it, vi } from "vitest";
+
+import { CreateNewPlanPage } from "@/app/plans/edit/[id]/page.client";
+import { SidebarProvider } from "@/components/ui/sidebar";
+
+import { departmentHandler, registrationsHandler } from "../mocks/handlers";
+import { server } from "../mocks/server";
+
+const mockUseSession = vi.fn(() => ({
+  user: null,
+}));
+
+vi.mock("@/hooks/use-session", () => ({
+  useSession: () => mockUseSession(),
+}));
+
+vi.mock("@/hooks/use-share", () => ({
+  useShare: () => ({
+    isDialogOpen: false,
+    openDialog: vi.fn(),
+    closeDialog: vi.fn(),
+    setIsDialogOpen: vi.fn(),
+  }),
+}));
```

Comment:
nie trzeba chyba tego mockować? mogłeś to jakoś ograć kodem, staraj się mockować jak najmniej